lib/khttp/ktest/credentials.go

```diff
@@ -0,0 +1,33 @@
+// +build !release
+
+package ktest
+
+import (
+	"errors"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+)
+
+//Check Credentials will walk runfiles until it reaches credentials.json and returns it
+func CheckCredentials() (string, error) {
```

Comment:
The reason why I took it from the runfiles dir and didn't glob() the file is because you cannot fetch files outside the $CWD and its children (by design). Also it is a data dependency, just an absolute path instead of a glob. 

https://github.com/aaahrens/enkit/blob/astore_delete/astore/server/astore/BUILD.bazel#L35

I wanted to avoid what currently happening which is creating several folders with different credential files. I think it's much simpler to have one top level folder for credentials and fetch on demand.

Reply:
ok, multiple orthogonal topics, let me try to break down the problem.

1) Regardless of agreeing or disagreeing with a global credentials directory vs one per folder being better, I would prefer to treat this as an orthogonal change. In this PR, I would use the same mechanism (separate credentials, glob()) that all other directories/subprojects are using. And then have a separate PR to change the mechanism, ideally for all projects at the same time, once we have something better ready.

Consider that we will need to re-factor this mechanism to allow for some sort of remote fetching of those credentials at build time - as right now credentials only live on my laptop. And if all use the same mechanism, it'll be easier to change.

2) The bazel way to refer to those files would be to create a rule for the top level credentials/ directory (eg, in the credentials directory itself, or parent directory). A `filegroup()` "packing" the `credentials/` for everyone to refer to. And then here in the BUILD file, just have a data dependency on that rule.

3) The credentials for different components are different - both in scope and configuration. As a habit, I think different components should have different credentials, scoped to allow the minimum set of things that the component is supposed to do.

Also, consider that some of the components we have are logically decoupled / separate, the fact that they are in the same repo is an implementation detail :)